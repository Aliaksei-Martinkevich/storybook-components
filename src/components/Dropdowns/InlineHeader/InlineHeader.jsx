import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './InlineHeader.scss';


const InlineHeader = ({ text, className = '' }) => (
  <span className={classNames({
    'dropdown-inline-header': true,
    [className]: true,
  })}
  >
    {text}
  </span>
  );

InlineHeader.propTypes = {
  text: PropTypes.string,
  className: PropTypes.string,
};

export default InlineHeader;
