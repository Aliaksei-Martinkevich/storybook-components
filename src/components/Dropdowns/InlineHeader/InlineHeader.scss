.text_dropdown-header {
    text-decoration-line: underline;
    text-decoration-style: dotted;
    &:hover {
        text-decoration-line: unset;
    }
    &.expanded {
        text-decoration-line: unset;
    }
}