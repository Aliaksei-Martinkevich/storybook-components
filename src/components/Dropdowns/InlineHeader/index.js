import React from 'react';
import PropTypes from 'prop-types';

import './index.css';


const InlineHeader = ({ text, className = ''}) => {
  return (
    <span className={`dropdown-inline-header ${className}`}>
      {text}
    </span>
  )
};

InlineHeader.propTypes = {
  text: PropTypes.string,
  className: PropTypes.string,
};

export default InlineHeader;
