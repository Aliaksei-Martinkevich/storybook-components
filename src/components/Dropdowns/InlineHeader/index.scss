.dropdown-inline-header {
    text-decoration-line: underline;
    text-decoration-style: dotted;
}

.dropdown-inline-header:hover {
    text-decoration-line: unset;
}

.expanded .dropdown-inline-header {
    text-decoration-line: unset;
}
