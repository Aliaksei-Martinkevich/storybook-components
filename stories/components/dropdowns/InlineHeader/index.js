import React from 'react';
import { storiesOf } from '@storybook/react';

import Components from '../../../../src/components'
const InlineHeader = Components.Dropdowns.InlineHeader;

export default storiesOf('Dropdown Inline Header', module)
  .add('Expanded', () => <div className='expanded'><InlineHeader text='Expanded header' /></div>)
  .add('Collapsed', () => <div className='collapsed'><InlineHeader text='Collapsed header' /></div>);