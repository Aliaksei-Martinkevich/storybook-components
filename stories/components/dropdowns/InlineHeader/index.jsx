import React from 'react';
import { storiesOf } from '@storybook/react';
import InlineHeader from '../../../../src/components/Dropdowns/InlineHeader/InlineHeader.jsx';

export default storiesOf('Dropdown Inline Header', module)
  .add('Expanded', () => <InlineHeader expanded text="Expanded header" />)
  .add('Collapsed', () => <InlineHeader text="Collapsed header" />);
