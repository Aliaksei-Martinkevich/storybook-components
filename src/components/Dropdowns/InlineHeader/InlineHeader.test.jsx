import React from 'react';
import renderer from 'react-test-renderer';
import InlineHeader from './InlineHeader.jsx';

describe('Inline header', () => {
  it('Should be collapsed', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const component = renderer.create(
      <InlineHeader
        className={customClassName}
        text={text}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be expanded', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const component = renderer.create(
      <InlineHeader
        className={customClassName}
        text={text}
        expanded
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
