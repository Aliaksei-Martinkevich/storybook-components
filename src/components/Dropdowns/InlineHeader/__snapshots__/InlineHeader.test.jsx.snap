// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline header Should be collapsed 1`] = `
<span
  className="text_dropdown-header class-name-example collapsed"
>
  Some text
</span>
`;

exports[`Inline header Should be expanded 1`] = `
<span
  className="text_dropdown-header class-name-example expanded"
>
  Some text
</span>
`;
